src/System.Private.Uri/tests/FunctionalTests/UriIpHostTest.cs

```diff
@@ -234,6 +234,31 @@ private void ParseBadIPv4Address(string badIpv4String)
 
         #region IPv6
 
+        [Theory]
+        [InlineData("fe80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("Fe80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("fE80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "%18")]
+        [InlineData("FE80::e077:c9a3:eeba:b8e9", "%eth10")]
```

Comment:
I think it would be nice to add some additional test cases here for more interesting scope IDs. What happens if we include an empty scope? Or scopes containing unusual characters?

Reply:
> What happens if we include an empty scope?

In this case what we want?Do we need to throw invalid address exception?Can we "parse" LLA without zone index?

Reply:
> Or scopes containing unusual characters?

With special chars like `[InlineData("FE80::e077:c9a3:eeba:b8e9", "\u30AF\u20E7")]` we get exception `Invalid URI: The hostname could not be parsed.` before reaching Ipv6 parser.

Reply:
Throwing an exception with special characters seems like the right behavior, especially considering we do the same thing before the change. 

On the other hand, throwing a new exception for an empty scope is risky. I think we could justify it if creating an LLA without a zone id really isn't a meaningful operation, but otherwise it's probably better to be more accepting of potentially valid addresses. As far as I can tell, the spec _implies_ that the zone id should be non-null, but never says it explicitly. From [RFC 4007 11.2](https://tools.ietf.org/html/rfc4007#section-11.2):
```
An implementation MAY support other kinds of non-null strings as <zone_id>.
```
So without the spec explicitly forbidding null zone ids, I'd be inclined to stick with the behavior you have in the PR now. @davidsh, any thoughts on this?

Reply:
>@davidsh, any thoughts on this?

This PR should be focussed on one thing.  And changing behavior for something else shouldn't be part of that.

So, no, don't start throwing a new exception for empty scope-id.

Reply:
@rmkerr @davidsh i added empty scope-id test case, would you like also a test with "special characters" for zone id(throwing exception assert)?

Reply:
It'd be nice to have that behavior pinned, so yes! That should be easy to implement using Assert.Throws. Also, could you try the case where we have the zone ID "%" with no scope ID following. It's another unusual case, but I would hope we get the same behavior as when we provide a totally empty zone ID.